activerecord/lib/active_record/core.rb

```diff
@@ -79,6 +79,8 @@ def self.configurations
 
       class_attribute :strict_loading_by_default, instance_accessor: false, default: false
 
+      class_attribute :allow_deprecated_singular_associations_name, instance_writer: false, default: true
```

Comment:
I still think we should make this a global on `::ActiveRecord`.

Reply:
I agree configuring this per class seems like a really bad idea. Updated thanks!